cmd/go.mod

```diff
@@ -34,5 +33,6 @@ require (
 	google.golang.org/api v0.0.0-20150730141719-0c2979aeaa5b
 	google.golang.org/appengine v1.6.5 // indirect
 	gopkg.in/olivere/elastic.v2 v2.0.12
+	k8s.io/apimachinery v0.18.2
```

Comment:
use https://github.com/kubernetes/utils/tree/master/clock instead of apimachinery.  There are some import cycle problems if we do that.  It is the same package, though

Reply:
There is an issue with such an import:
```
➜  cadvisor git:(improved_prometheus_tests) ✗ make test
go: finding module for package github.com/kubernetes/utils/clock
go: found github.com/kubernetes/utils/clock in github.com/kubernetes/utils v0.0.0-20200414100711-2df71ebbae66
go: github.com/google/cadvisor/metrics imports
        github.com/kubernetes/utils/clock: github.com/kubernetes/utils@v0.0.0-20200414100711-2df71ebbae66: parsing go.mod:
        module declares its path as: k8s.io/utils
                but was required as: github.com/kubernetes/utils
go: finding module for package github.com/kubernetes/utils/clock
go: found github.com/kubernetes/utils/clock in github.com/kubernetes/utils v0.0.0-20200414100711-2df71ebbae66
go: github.com/google/cadvisor/cmd/internal/http imports
        github.com/kubernetes/utils/clock: github.com/kubernetes/utils@v0.0.0-20200414100711-2df71ebbae66: parsing go.mod:
        module declares its path as: k8s.io/utils
                but was required as: github.com/kubernetes/utils
>> running tests
?       github.com/google/cadvisor      [no test files]
go: finding module for package github.com/kubernetes/utils/clock
go: found github.com/kubernetes/utils/clock in github.com/kubernetes/utils v0.0.0-20200414100711-2df71ebbae66
go: github.com/google/cadvisor/cmd imports
        github.com/google/cadvisor/cmd/internal/http imports
        github.com/kubernetes/utils/clock: github.com/kubernetes/utils@v0.0.0-20200414100711-2df71ebbae66: parsing go.mod:
        module declares its path as: k8s.io/utils
                but was required as: github.com/kubernetes/utils
make: *** [Makefile:32: test] Error 1
```

Reply:
Are you importing it as "k8s.io/utils/clock" or "github.com/kubernetes/utils/clock"?  You should be using the k8s.io path.

We already use k8s.io/utils/clock in manager/container.go, for example

Reply:
Take a look at d8072ed104d11c7e774a5066048edae75186b332. When I run `go mod tidy` I get following error:
```
➜  cadvisor git:(improved_prometheus_tests) ✗ go mod tidy
go: finding module for package k8s.io/utils/mount
go: finding module for package github.com/kubernetes/utils/clock
go: finding module for package github.com/google/uuid
go: finding module for package k8s.io/utils/inotify
go: finding module for package k8s.io/utils/clock/testing
go: finding module for package k8s.io/utils/clock
go: found k8s.io/utils/inotify in k8s.io/utils v0.0.0-20200414100711-2df71ebbae66
go: found github.com/kubernetes/utils/clock in github.com/kubernetes/utils v0.0.0-20200414100711-2df71ebbae66
go: found github.com/google/uuid in github.com/google/uuid v1.1.1
go: github.com/google/cadvisor/metrics imports
        github.com/kubernetes/utils/clock: github.com/kubernetes/utils@v0.0.0-20200414100711-2df71ebbae66: parsing go.mod:
        module declares its path as: k8s.io/utils
                but was required as: github.com/kubernetes/utils
```